concepts/errors-handling/introduction.md

```diff
@@ -1,3 +1,70 @@
 # Introduction
 
-TODO: add introduction for errors-handling concept
+In Go, errors must be the last value returned in a multiple returns function.
+
+Always return the default value for the type, if returning an error:
+
+```go
+func Foo() (int, error) {
+  // Ok!
+  return 0, errors.New("Error")
+
+  // Don't
+  return 10, errors.New("Error")
+}
+```
+
+When no errors have to be returned, return `nil`:
+
+```go
+func Foo() (int, error) {
+  return 10, nil
+}
+```
+
+Errors can be checked against `nil`:
+
+```go
+file, err := os.Open("./users.csv")
+if err != nil {
+    // do something with err
+    return // or continue
+}
+// do something with file
+```
+
+It is best practise to stop the execution when an error is encountered rather
+than using `else`. The example below shows a bad practice:
+
+```go
+file, err := os.Open("./users.csv")
+if err != nil {
+  // do something wit err
+} else {
+  // do something with file
+}
+```
+
+## Scope of error
+
+In Go, it is also a best practice to avoid declaring variables in a scope if
+these are not required later.
+
+For the sake of this example, we want to check for errors only:
```

Comment:
> For the sake of this example, we want to check for errors only:

Would it be better, for simplicity's sake perhaps, to use a function that returns only an error in this example and avoid having to explain why we are not capturing the first return value? For example `os.Rename()`.